mount/mount_linux.go

```diff
@@ -63,7 +63,17 @@ func (m *Mount) Mount(target string) error {
 	if flags&unix.MS_REMOUNT == 0 || data != "" {
 		// Initial call applying all non-propagation flags for mount
 		// or remount with changed data
-		if err := mountAt(chdir, m.Source, target, m.Type, uintptr(oflags), data); err != nil {
+		source := m.Source
+		if losetup {
+			dev, devFile, err := setupLoop(m.Source, oflags&unix.MS_RDONLY == unix.MS_RDONLY)
```

Comment:
How are these removed?

Reply:
Please see man(4) loop:
```
              LO_FLAGS_AUTOCLEAR (since Linux 2.6.25)
                     The loopback device will autodestruct on last close.
```

By setting the `loFlagsAutoclear` flag, the loopback device is automatically removed in two situations:
1. If mount fails, it is removed when devFile is closed;
2. If mount succeeds and devFile is closed, it is removed when it is unmounted.

The behavior is also documented in the comments above `setupLoop()`. And it is the reason why I need to amend `mount.All` to support `-o loop` option because the `losetup` binary cannot support `loFlagsAutoclear`.